### PR TITLE
Annotate WebAssembly.Authentication for nullability

### PR DESCRIPTION
**Pull request by JamesNK**
Addresses https://github.com/dotnet/aspnetcore/issues/5680

### CODE REVIEW

File: src/Components/WebAssembly/WebAssembly.Authentication/src/NavigationManagerExtensions.cs [hunk 1]
```diff
@@ -30,14 +30,14 @@ public static class NavigationManagerExtensions
...
-    public static void NavigateToLogout(this NavigationManager manager, string logoutPath, string returnUrl)
+    public static void NavigateToLogout(this NavigationManager manager, string logoutPath, string? returnUrl)
     {
         manager.NavigateTo(logoutPath, new NavigationOptions
         {
             HistoryEntryState = new InteractiveRequestOptions
             {
                 Interaction = InteractionType.SignOut,
-                ReturnUrl = returnUrl
+                ReturnUrl = returnUrl ?? manager.Uri
```
Comment (JamesNK) [inline src/Components/WebAssembly/WebAssembly.Authentication/src/NavigationManagerExtensions.cs:40]: I'm not sure about this change. Can `InteractiveRequestOptions.ReturnUrl` be null?

**Comment by JamesNK** [comment]
@dotnet/aspnet-build How do I fix the public API files.

I tried updating just the Unshipped.txt. As an example:

```
*REMOVED*~Microsoft.Extensions.DependencyInjection.IRemoteAuthenticationBuilder<TRemoteAuthenticationState, TAccount>
*REMOVED*~Microsoft.Extensions.DependencyInjection.IRemoteAuthenticationBuilder<TRemoteAuthenticationState, TAccount>.Services.get -> Microsoft.Extensions.DependencyInjection.IServiceCollection

Microsoft.Extensions.DependencyInjection.IRemoteAuthenticationBuilder<TRemoteAuthenticationState, TAccount>
Microsoft.Extensions.DependencyInjection.IRemoteAuthenticationBuilder<TRemoteAuthenticationState, TAccount>.Services.get -> Microsoft.Extensions.DependencyInjection.IServiceCollection!
```

But I got warnings that the type exists twice. How do I fix this?

> Warning	RS0025	The symbol 'Microsoft.Extensions.DependencyInjection.IRemoteAuthenticationBuilder<TRemoteAuthenticationState, TAccount>' appears more than once in the public API files	Microsoft.AspNetCore.Components.WebAssembly.Authentication	C:\Development\Source\aspnetcore\src\Components\WebAssembly\WebAssembly.Authentication\src\PublicAPI.Unshipped.txt	32	Active

**Comment by BrennanConroy** [comment]
Update the shipped.txt file in this case. The analyzer doesn't work well when going from non-nullable to nullable APIs.

**Comment by JamesNK** [comment]
@aspnet-build How do I make the build pass with the shipped.txt changes?

**Comment by BrennanConroy** [comment]
> How do I make the build pass with the shipped.txt changes?

You don't. Someone with permissions will need to merge.

**Comment by JamesNK** [comment]
@aspnet-blazor-eng Please review

**Comment by JamesNK** [comment]
@mkArtakMSFT

**Comment by mkArtakMSFT** [comment]
Thanks @JamesNK.
@MackinnonBuck can you please review this? Thanks!

File: src/Components/WebAssembly/WebAssembly.Authentication/src/NavigationManagerExtensions.cs (hunk 1 above)
Comment (MackinnonBuck) [reply src/Components/WebAssembly/WebAssembly.Authentication/src/NavigationManagerExtensions.cs]: It seems like we assume the return URL can be `null` elsewhere in `WebAssembly.Authentication`, e.g. [`RemoteAuthenticatorViewCore`](https://github.com/dotnet/aspnetcore/blob/a80400004a4b2e5c0d3f9cff67641d0a47b0202c/src/Components/WebAssembly/WebAssembly.Authentication/src/RemoteAuthenticatorViewCore.cs#L370). Maybe we can make `InteractiveRequestOptions.ReturnUrl` nullable and not required?

cc @javiercn

**Review by MackinnonBuck — COMMENTED** [review]
This looks good to me. I think [this](https://github.com/dotnet/aspnetcore/pull/43442#discussion_r986102072) is the main question that we still need to make a decision on. This can be merged once we resolve that.

@javiercn do you have any thoughts on this?

**Comment by JamesNK** [comment]
I'd like to be able to merge this. Can the SME please review and approve/require changes 🙏

**Comment by javiercn** [comment]
@JamesNK let me take a look.

**Comment by JamesNK** [comment]
@javiercn ping

File: src/Components/WebAssembly/WebAssembly.Authentication/src/Models/InteractiveRequestOptions.cs [hunk 2]
```diff
@@ -58,7 +58,7 @@ public bool TryRemoveAdditionalParameter(string name)
     /// <summary>
     /// Tries to retrieve an existing additional parameter.
     /// </summary>
-    public bool TryGetAdditionalParameter<[DynamicallyAccessedMembers(JsonSerialized)] TValue>(string name, out TValue value)
+    public bool TryGetAdditionalParameter<[DynamicallyAccessedMembers(JsonSerialized)] TValue>(string name, out TValue? value)
```
Comment (javiercn) [inline src/Components/WebAssembly/WebAssembly.Authentication/src/Models/InteractiveRequestOptions.cs:61]: Do we need to apply here the Attribute `[NotNullWhenTrue]` for the out parameter?

**Review by javiercn — APPROVED** [review]
One small comment, otherwise looks good.

**Comment by japui-coder** [comment]
Annotate WebAssembly.Authentication for nullability #43442

**Review by japui-coder — CHANGES_REQUESTED** [review]
...Components/WebAssembly/WebAssembly.Authentication/src/Models/InteractiveRequestOptions.cs

**Pull request merged**